core/src/main/java/com/oath/oak/OakNativeMemoryAllocator.java

```diff
@@ -195,9 +195,7 @@ public boolean isClosed() {
     @Override
     public void readByteBuffer(Slice s) {
         Block b = blocksArray[s.getAllocatedBlockID()];
-        // The returned buffer is this thread's block buffer.
-        // Therefore, a thread cannot read two slices from the same block without duplicating one of them.
-        b.getBufferForThread(s);
+        b.getBuffer(s);
```

Comment:
The opposite order looks more reasonable to me. Like slice.setBuffer(b.getBuffer)... Currently the way it works is obscure and looks like block is getting a buffer from slice, from the first glance...

Reply:
Note that `Block` has two buffers. So what you are suggesting is actually:
```java
s.setBuffer(b.getReadBuffer(), b.getWriteBuffer())
````
And this line would repeat twice. Once in `OakNativeMemoryAllocator.readByteBuffer(Slice)`, and once in `Block.allocate(Slice, int)`.

Your suggestion also breaks the convention that buffers are never exposed directly, but rather passed using a `Slice` objects.

If it looked obscure it might be because of the method name `Block.getBuffer(Slice)`.
How about using the same name from `OakBlockMemmoryAllocator`: `void readByteBuffer(Slice s)`?

Reply:
Yes, at least rename it

Reply:
done